template/alpha_formula.py

```diff
@@ -23,24 +24,29 @@ def __init__(self, _startdate, data, parameters):
         self.mid_last_k_days = {
             instrument: deque(
                 (datum.buy + datum.sell) / 2
-                for datum in price_data[-self.k_days - 1: -1])
+                if datum.buy is not None and datum.sell is not None else 0
```

Comment:
The price may be None, so do this change.

Reply:
`if all(datum.buy, datum.sell) else 0`

Reply:
To make the change easier, we can let this be empty at first, and add a check in the update part, which checks whether `len(self.mid_last_k_days) < k`. If true, just append the new datum without popping anything; otherwise, do what our current code does.